rfcs/013-release_deployment_tracking/README.md

```diff
@@ -4,396 +4,460 @@
+- **release:** Metadata indicating the intention to deploy a particular **build artifact** at a given **release hash**. Generally part of a **release set**.
 
-The attributes are described as follows:
-- **project_id**: an identifier for a project.
-- **label:** A label for the set of images released, e.g. latest, stable, v2-branch
+- **release set:** A set of **build artifacts** at particular **release hashes** based on a **service set** template that is intended to be released into an **environment** together.
 
- For example:
+- **deployment:** A deployed **service**.
 
- `/storage/images/latest/archivist`
+- **environment:** Where you deploy your **release sets** when you want them to run e.g. staging, production.
 
-SSM parameters will provide a versioned record of build artifacts. SSM allows descriptions to be added to updates, these descriptions should contain the `user_id` of what or who is updating the version
+- **deployment set:** A set of deployed **services** created from a **release set** that has been deployed into an **environment**.
 
-This mechanism will be provided via a python application, packaged in a docker container distributed via the https://github.com/wellcometrust/dockerfiles repository.
+#### How these terms fit together
 
-### Project structure
+![Terms](terms.png)
 
-> **Project:** A set of services that when composed perform a function.
+## What we do now
 
-In order to build releases that describe which version of a service to deploy to a particular environment we need a machine readable description of project structure.
+See the documentation on [version 1](v1/README.md).
 
-We propose the following structure:
+### Problems with the current approach
 
-```json
-{
-  "project": {
-    "id": "id",
-    "name": "project_name",
-    "profile": "aws_profile"
-  },
-  "environments" : [
-    {
-      "id": "environment_id",
-      "name": "environment_name"
-    }
-  ]
-}
+- It is not clear how to release a single service
+- In order to actually deploy something there are multiple steps:
+  - Create a release set using the CLI tool
+  - Deploy a release set  using the CLI tool
+  - Run `terraform apply` to _actually_ update the running services
+- Release/Deploy descriptions are not well used / hidden
+- Poor visibility of what is actually deployed
+
+#### Moving away from terraform for deployment
+
+We currently use `terraform apply` to deploy services at a particular release hash. The choice to use terraform was driven by a requirement to describe our task definitions in code. 
+
+Separating service deployment from infrastructure changes is desirable as infra/service deployments have differing concerns and pace, i.e. high-value infrequent (infra), vs. low-value frequent (deploying new versions of services). 
+
+Running terraform in a CI environment like Travis is not desirable as giving an automated environment the power to run infrastructure updates needs careful consideration.
```

Comment:
The point of writing RFC is to give careful consideration to these things!

We have discussed reasons we don't want to give Travis credentials:

* risk of credentials being leaked or misused
* we don't usually have permanent credentials
* we've found an alternative

And possibly others I've forgotten.

Also, use more active voice. “We don’t want to give Travis the power to make infrastructure changes because XYZ”, not “it is not desirable”. It's more direct, and we're stating our opinion, not an objective fact.